src/core/globalstate.ts

```diff
@@ -45,6 +45,12 @@ export class MobXGlobals {
 	isRunningReactions = false;
 
 	/**
+	 * Current reaction (if any is running).
+	 * Reactions are run after derivations using a trampoline.
+	 */
+	currentReaction: Reaction = undefined;
```

Comment:
Instead of introducing a separate state var it is probably simpler to just peek into the `derivationStack` and check whether it is an instanceof Reaction